src/Domain/CreateInitialElement.hpp

```diff
@@ -22,14 +24,13 @@ namespace Initialization {
  *
  * \details This function creates an element at the refinement level and
  * position specified by the `element_id` within the `block`. It assumes
- * that all of its neighboring elements are at the same refinement level. Thus,
- * the created element's `neighbors` has one neighbor per direction (or zero on
- * an external boundary), with each neighbor at the same refinement level as the
- * element.
+ * that all elements in a given block have the same refinement level,
+ * given in `initial_refinement`.
  */
 template <size_t VolumeDim>
 Element<VolumeDim> create_initial_element(
-    const ElementId<VolumeDim>& element_id,
-    const Block<VolumeDim>& block) noexcept;
+    const ElementId<VolumeDim>& element_id, const Block<VolumeDim>& block,
+    const std::vector<std::array<size_t, VolumeDim>>&
+        initial_refinement) noexcept;
```

Comment:
[optional] would `initial_refinement_levels` be more descriptive? I'm personally a fan of `refinement` -> `refinement_lev(el)` since `refinement` alone makes it unclear of talking about `2^L` or `L`. I'm also fine saying we want this change and filing an issue to do it at a later time if that's what you prefer :)